recipe/meta.yaml

```diff
@@ -2,6 +2,16 @@
 {% set version = "2.5.0" %}
 {% set sha256 = "ff72d21ddf62d829f43375c67ea4934c82947f9a0c5cbf9114838624db941420" %}
 
+# We are overriding the default numpy pins slightly because statsmodels
+# was not built against 1.16 in all cases it was supposed to be. We are
+# doing it here instead of with a custom conda_build_config.yaml because
+# the latter is apparently incompatible with our prefect build process,
+# silently skipping one or more of the python versions.
+{% set numpy = "1.16" %}  # [py<39 and not (aarch64 or s390x or arm64)]
+{% set numpy = "1.19" %}  # [py<39 and (aarch64 or s390x or arm64)]
+{% set numpy = "1.19" %}  # [py==39]
+{% set numpy = "1.21" %}  # [py>39]
```

Comment:
The reason we need to adjust the numpy pins is that our `statsmodels` package was built against numpy 1.19 in certain cases where our default pin is 1.16. That means that any package that depends on statsmodels must inherit those non-standard pins. My original solution was to deliver an alternative set of `python`, `numpy`, and `zip_keys` entries in a custom `conda_build_config.yaml` file. For reasons none of us seem to fully understand, this was causing some of the builds to be skipped.

The solution to use jinja pins instead was recommended by @cbouss, who shared the example from  our scipy feedstock:

https://github.com/AnacondaRecipes/scipy-feedstock/blob/master/recipe/meta.yaml#L2-L4

So you can confirm the logic is the same, here is the contents of the deleted conda_build_config.yaml

```
numpy:
- 1.16 # [not (aarch64 or s390x or arm64)]
- 1.16 # [not (aarch64 or s390x or arm64)]
- 1.19 # [aarch64 or s390x]
- 1.19 # [aarch64 or s390x or arm64]
- 1.19
- 1.21
python:
- 3.7  # [not arm64]
- 3.8
- 3.9
- "3.10"
zip_keys:
- numpy
- python
```